doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,324 @@
+  have a prefix in that list, then one must be created.
+
+.. note::
+
+  The SoC directory name does not need to match the name of the SoC.
+  Multiple SoC can even defined be in one directory. In Zephyr, SoCs are often
+  organized in sub-folders under a common SoC Family or SoC Series tree.
+
+Your SoC directory should look like this:
+
+.. code-block:: none
+
+   soc/<VENDOR>/<soc-name>
+   ├── soc.yml
+   ├── soc.h
+   ├── CMakeLists.txt
+   ├── Kconfig
+   ├── Kconfig.soc
+   └── Kconfig.defconfig
+
+Replace ``<soc-name>`` with your SoC's name.
+
+
+The mandatory files are:
+
+#. :file:`soc.yml`: a YAML file describing the high-level meta data of the
+   SoC such as:
+   - SoC name: the name of the SoC name
+   - CPU clusters: CPU clusters if the SoC contains one or more clusters
+   - SoC series: the SoC series to which the SoC belong
+   - SoC family: the SoC family to which the series belong
+
+#. :file:`soc.h`: a header file which can be used to describe or provide
+   configuration macros for the SoC. The :file:`soc.h` will often be included in
+   drivers, sub-systems, boards, and other source code found in Zephyr.
+
+#. :file:`Kconfig.soc`: the base SoC configuration which defines a Kconfig SoC
+   symbol and provides the SoC name to the Kconfig ``SOC`` setting.
+   If the ``soc.yml`` describes a SoC family and series, then those must also
+   be defined in this file. Kconfig settings outside of the SoC tree must not be
+   selected. To select general Zephyr Kconfig settings the :file:`Kconfig` file
+   must be used.
+
+#. :file:`CMakeLists.txt`: CMake file loaded by the build system. This CMake
+   file can define additional include paths and/or source files to be used when
+   a build targets the SoC. Also the base line linker script to use must be
+   defined.
+
+The optional files are:
+
+- :file:`Kconfig`, :file:`Kconfig.defconfig` software configuration in
+  :ref:`kconfig` format. These select the architecture and peripherals
+  available.
+
+Write your SoC YAML
+*********************
+
+The SoC YAML file describes the SoC family, SoC series, and SoC at a high level.
+
+Detailed configurations, such as hardware description and configuration are done
+in devicetree and Kconfig.
+
+The skeleton of a simple SoC YAML file containing just one SoC is:
+
+.. code-block:: yaml
+
+   socs:
+   - name: <soc1>
+
+It is possible to have multiple SoC located in the SoC folder.
+For example if they belong to a common family or series it is recommended to
+locate such SoC in a common tree.
+Multiple SoCs and SoC series in a common folder can be described is the
+:file:`soc.yml` file as:
+
+.. code-block:: yaml
+
+   family:
+     name: <family-name>
+     series:
+     - name: <series-1-name>
+       socs:
+       - name: <soc1>
+         cpucluster:
+         - name: <coreA>
+         - name: <coreB>
+           ...
+       - name: <soc2>
+     - name: <series-2-name>
+       ...
+
+
+Write your devicetree
```

Comment:
Write your devicetree
->
Write your SOC devicetree description
maybe?

Reply:
We are inside the SoC porting guide 😉 
But I can add it if that's preferred.

I'm happy both with and without the `SoC`.